distributed/client.py

```diff
@@ -5445,7 +5456,7 @@ async def _wait(fs, timeout=None, return_when=ALL_COMPLETED):
         {fu for fu in fs if fu.status != "pending"},
         {fu for fu in fs if fu.status == "pending"},
     )
-    cancelled = [f.key for f in done if f.status == "cancelled"]
+    cancelled = {f.key: f._state.exception for f in done if f.cancelled()}
```

Comment:
This might become very large. Ideally, we'd group these by the reason for cancellation.